### ISSUE TITLE
[Feature Request] Docker swarm mode

### ISSUE THREAD

**Issue by SophiaHadash** [benphelps/homepage#348]
**Describe the bug**
When you have multiple docker nodes in docker swarm mode, you cannot use the `container_name` property in your compose file. Instead, docker swarm will use container names such as `/nextcloud_nextcloud.1.n1ykvwhp0cbkfxwpton2uq6jj`. I can use this container name in my configuration, but the problem is that the name will change when the service is restarted or updated. 

**Expected behavior**
It seems natural that users that have multiple docker servers have them connected using swarm or kubernetes. I think it makes sense to support docker swarm as it is mostly used for smaller servers.

A possible solution would be to look for the service name instead. In my case, this is `nextcloud_nextcloud`. Then, on the swarm manager I can get a container id like this:

`docker ps --no-trunc | grep -E 'nextcloud_nextcloud\.' | awk '{print $1;}'`

On swarm workers this is slightly more complicated, but I think still doable. 

**Additional context**
Official docker swarm doc:

```
Note when using docker stack deploy
The [container_name][1] option is ignored when [deploying a stack in swarm mode][1]
```

Please let me know what you think about swarm support. I might be able to allocate some time, but I cannot guarantee it as of now.

**Comment by benphelps** [comment]
Thanks for reporting this, I'm not familiar with Docker swarm, but the current Docker support is strictly tied to Docker alone, swarm support would be a pretty large undertaking.

I'll change this from Bug to Feature Request, as homepage isn't designed to work with Docker Swarm

**Comment by shamoon** [comment]
Closed by #660
